common/src/main/java/com/wynntils/features/user/redirects/ChatRedirectFeature.java

```diff
@@ -86,6 +91,16 @@ private void register(Redirector redirector) {
         redirectors.add(redirector);
     }
 
+    // Data structure is String Message stores Entry of MessageContainer which stores Integers for the amount of times
+    // the message has been sent and the associated time (Long) in milliseconds
+    private Map<String, Map.Entry<MessageContainer, Map.Entry<Integer, Long>>> cachedNotifications = new HashMap<>();
```

Comment:
Cant you use a Set for this? If you can, use TimesSet, so we don't fill the memory with cache, as @magicus pointed out